src/Domain/FunctionsOfTime/PiecewisePolynomial.hpp

```diff
@@ -52,13 +52,29 @@ class PiecewisePolynomial : public FunctionOfTime {
   /// If `MaxDeriv` is 1 and `update` has been called for time `t`, the updated
   /// value is ignored.
   std::array<DataVector, 2> func_and_deriv(double t) const override {
-    return func_and_derivs<1>(t);
+    if constexpr (MaxDeriv > 0) {
+      return func_and_derivs<1>(t);
+    } else {
+      const auto tmp = func_and_derivs<0>(t);
+      const size_t size = tmp[0].size();
+      return {{tmp[0], DataVector(size, 0.0)}};
```

Comment:
Could you add some comments to explain what's going on here? Why does this break only on M1?

Reply:
Actually, I may not need these changes, I just need to change what is instantiated; I was worried `func_and_derivs` didn't work if you ask for too many derivs because those weren't explicitly instantiated, but it looks like `func_and_derivs` works correctly in that case; I'll check/update the test to make sure it does.

M1 appears to not instantiate a function called by another explicitly instantiated function.  That was the case for the other instantiations I had to add.

 ```
Undefined symbols for architecture arm64:
  "std::__1::array<DataVector, (1ul) + (1)> domain::FunctionsOfTime::PiecewisePolynomial<0ul>::func_and_derivs<1ul>(double) const", referenced from:
      domain::FunctionsOfTime::PiecewisePolynomial<0ul>::func_and_deriv(double) const in libTest_ControlSystem.a(Test_Trigger.cpp.o)
      domain::FunctionsOfTime::PiecewisePolynomial<0ul>::func_and_deriv(double) const in libTest_FunctionsOfTime.a(Test_PiecewisePolynomial.cpp.o)
      domain::FunctionsOfTime::PiecewisePolynomial<0ul>::func_and_deriv(double) const in libTest_ControlSystem.a(Test_InitializeMeasurements.cpp.o)
  "std::__1::array<DataVector, (2ul) + (1)> domain::FunctionsOfTime::PiecewisePolynomial<0ul>::func_and_derivs<2ul>(double) const", referenced from:
      domain::FunctionsOfTime::PiecewisePolynomial<0ul>::func_and_2_derivs(double) const in libTest_ControlSystem.a(Test_Trigger.cpp.o)
      domain::FunctionsOfTime::PiecewisePolynomial<0ul>::func_and_2_derivs(double) const in libTest_FunctionsOfTime.a(Test_PiecewisePolynomial.cpp.o)
      domain::FunctionsOfTime::PiecewisePolynomial<0ul>::func_and_2_derivs(double) const in libTest_ControlSystem.a(Test_InitializeMeasurements.cpp.o)
ld: symbol(s) not found for architecture arm64
```